src/Core/Services/StateService.cs

```diff
@@ -1708,6 +1710,21 @@ public async Task SetShouldConnectToWatchAsync(bool shouldConnect, string userId
                 ReconcileOptions(new StorageOptions { UserId = userId }, await GetDefaultStorageOptionsAsync());
             var key = Constants.ShouldConnectToWatchKey(reconciledOptions.UserId);
             await SetValueAsync(key, shouldConnect, reconciledOptions);
+            await UpdateLastUserShouldConnectToWatchAsync(shouldConnect);
+        }
+
+        public async Task<bool> GetLastUserShouldConnectToWatchAsync()
+        {
+            var reconciledOptions = await GetDefaultStorageOptionsAsync();
```

Comment:
Since there's no userId passed in here, there's nothing to reconcile, so the name should just be `options` (see the `PushRegisteredTokenAsync` getter/setter for reference)